### DISCUSSION TITLE
online installer

### DISCUSSION THREAD

**Discussion by PepiBobanov** [oleg-shilo/wixsharp#1352] [General]
Greetings.
I saw .msi unpacks before install so ...
Is it possible to make installer so, that i download required files before install , and then start installer?

**Comment by oleg-shilo** [comment]
MSI is a single file (MSI database file) that is self-sufficient and does not download anything from online resources. 
Thus there is no actually "downloading required files".

Though in the custom actions of your msi it may indeed access network resources for some validation (e.g. check the licence key). So it depends.

A bootstrapper (msi bundle) is a completely different story. Depending on the deployment strategy it can download all msi files and then do the installation. Though asked about downloading something for msi not the msi itself...

**Reply by PepiBobanov** [reply]
I understand.
I made system for updating files. And was wondering, if i could tell MSI to get current files from that system. 
But in this case i should consider to make some custom installer.

**Reply by oleg-shilo** [reply]
agree. it does not seem to follow the MSI deployment paradigm
